libkmod/libkmod-module.c

```diff
@@ -511,8 +511,10 @@ KMOD_EXPORT int kmod_module_new_from_name_lookup(struct kmod_ctx *ctx,
 
 KMOD_EXPORT int kmod_module_unref_list(struct kmod_list *list)
 {
-	for (; list != NULL; list = kmod_list_remove(list))
```

Comment:
humn... what's the problem with this line? if list is NULL it will simply return, like the change you did